src/main/java/com/singh/astha/notification/service/dtos/response/wrapper/ErrorResponse.java

```diff
@@ -2,28 +2,42 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.singh.astha.notification.service.enums.ErrorCode;
-import lombok.*;
+import lombok.Builder;
 
 import java.io.Serializable;
 
-@Getter
-@Setter
 @Builder
-@AllArgsConstructor(access = AccessLevel.PROTECTED)
 @JsonInclude(JsonInclude.Include.NON_NULL)
-public class ErrorResponse implements Serializable {
-    private String errorCode;
+public record ErrorResponse(String errorCode, String message, String detail, String help) implements Serializable {
 
-    private String message;
+    public static ErrorResponse from(ErrorCode errorCode) {
+        return ErrorResponse.builder()
+                .errorCode(errorCode.getCode())
+                .message(errorCode.getMessage())
+                .build();
+    }
 
-    private String detail;
+    public static ErrorResponse from(String message) {
+        return ErrorResponse.builder()
+                .message(message)
+                .build();
+    }
```

Comment:
Remove this

We should never allow someone creating an ErrorResponse without specifying the ErrorCode.